db/migrate/20181001131632_add_conversion_host_id_to_miq_request_tasks.rb

```diff
@@ -0,0 +1,46 @@
+class AddConversionHostIdToMiqRequestTasks < ActiveRecord::Migration[5.0]
+  class MiqRequestTask < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    serialize :options, Hash
+    belongs_to :conversion_host, :class_name => "AddConversionHostIdToMiqRequestTasks::ConversionHost"
+  end
+
+  class ConversionHost < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    belongs_to :resource, :polymorphic => true
+  end
+
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    add_column :miq_request_tasks, :conversion_host_id, :bigint
+
+    MiqRequestTask.where(:type => 'ServiceTemplateTransformationPlanTask').each do |task|
+      host_id = task.options[:transformation_host_id]
+      next unless host_id
+      host = Host.find_by(:id => host_id)
+      task.options.delete(:transformation_host_id)
+      if host.present?
+        task.conversion_host = ConversionHost.find_or_create_by!(:resource => host) do |ch|
+          ch.name                     = host.name
+          ch.vddk_transport_supported = true
+          ch.ssh_transport_supported  = false
+        end
+      end
+      task.save!
+    end
+  end
+
+  def down
+    conversion_host_ids = MiqRequestTask.where(:type => 'ServiceTemplateTransformationPlanTask').map do |task|
+      task.options[:transformation_host_id] = task.conversion_host.resource.id
+      task.save!
+      task.conversion_host.id
+    end
```

Comment:
Should `.uniq` also since there will be many tasks with the same conversion host.  You can do `end.uniq.compact` or `ConversionHost.destroy(conversion_host_ids.uniq.compact)` your call